crates/rust-analyzer/src/global_state.rs

```diff
@@ -69,6 +69,7 @@ pub(crate) struct GlobalState {
     pub(crate) config: Config,
     pub(crate) analysis_host: AnalysisHost,
     pub(crate) diagnostics: DiagnosticCollection,
+    pub(crate) completion_resolve_data: Arc<FxHashMap<usize, ImportEditPtr>>,
```

Comment:
I think we should try to handle this in a stateless manner, similar to how we tackle code actions. 

See https://github.com/rust-analyzer/rust-analyzer/commit/3e00bfce2b1f53b14ec89ddb249364eac23f5b1e for explanation.

For completion, I think we should use original completion request as a *data* field of the completion item, plus some tag to identify a specific thing we want to export.

Reply:
ah, I am just repeating what @kjeremy said apparently :D